src/Combinatorics/Matroids/matroid_strata_grassmannian.jl

```diff
@@ -0,0 +1,419 @@
+    
+    S = MatrixSpace(R, d, n-d)
+    X = S()
+    for j in 1:n-d, i in 1:d
+        if [i,j] in MC
+            X[i,j] = xdict[[i,j]]
+        else
+            X[i,j] = R(0)
+        end
+    end
+    return X
+end
+
+
+# M and N have same number of rows, and M has #B columns, both have entries in ring R
+function interlace_columns(M::MatrixElem{T}, N::MatrixElem{T}, B::Vector{Int},
+                           R::MPolyRing, x::Vector{T}) where T <: MPolyElem 
+    
+    M_nrows, M_ncols = size(M)
+    N_nrows, N_ncols = size(N)
+    n = M_ncols + N_ncols
+
+    S = MatrixSpace(R, M_nrows, n)
+    Bc = [i for i in 1:n if !(i in B)]
+    
+    X = S()
+    X[:, B] = M
+    X[:, Bc] = N
+    
+    return X 
+end
+
+# This makes the matrix X from which we compute the coordinate ring of the matroid
+# stratum. It has the identity matrix at columns indexed by B, 0's at locations
+# determined by the nonbases of X. 
+function make_coordinate_matrix(d::Int, n::Int, MC::Vector{Vector{Int}},
+                                B::Vector{Int},
+                                R::MPolyRing, x::Vector{T},
+                                xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    Id = identity_matrix(R,d)
+    Xpre = make_coordinate_matrix_no_identity(d, n, MC, R, x, xdict)
+    return interlace_columns(Id, Xpre, B, R, x)
+end
+
+
+# This function returns all d x d determinants of the matrix X from above
+# of all collections of d-columns coming from the bases of the matroid.
+
+function bases_determinants(X::MatrixElem{T}, Bs::Vector{Vector{Int}}) where {T<:MPolyElem}
+    return unique!([det(X[:, b]) for b in Bs ])
+end
+
+
+#function bases_determinants(X::Matrix{T}, Bs::Vector{Vector{Int}})  where {T<:MPolyElem}
+
+    #d::Int, n::Int, Bs::Vector{Vector{Int}},
+    #MC::Vector{Vector{Int}},
+    #B::Vector{Int}, R::MPolyRing, x::Vector{T},
+    #xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    #X = make_coordinate_matrix(d, n, MC, B, R, x, xdict)
+    
+#    return unique!([det(X[:, b]) for b in Bs ])
+#end
+
+
+# This forms the semigroup of the polynomial ring from make_polynomial_ring
+# generated by the determinants from bases_determinants.
+# function localizing_semigroup(d::Int, n::Int, Bs::Vector{Vector{Int}},
+#                               MC::Vector{Vector{Int}}, B::Vector{Int},
+#                               R::MPolyRing, x::Vector{T},
+#                               xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+
+    
+#     basesX = bases_determinants(X,Bs)
+#     #basesX = bases_determinants(d, n, Bs, MC, B, R, x, xdict)
+    
+#     sTotal = MPolyPowersOfElement(basesX[1])
+    
+#     if length(basesX) == 1
+#         return sTotal
+#     end
+    
+#     for i in 2:length(basesX)
+#         if !(basesX[i] in sTotal)
+#             sTotal = product(sTotal, MPolyPowersOfElement(basesX[i]))
+#         end
+#     end
+    
+#     return sTotal
+    
+# end
```

Comment:
Is this deprecated?

Reply:
I don't know the precise meaning of deprecated in this setting. But it is no longer used/useful given that this can be done using  `MPolyPowersOfElement`

Reply:
Ok, I vaguely remember. Then we might consider removing these lines. I know I have plenty of such stuff leftovers in my code, but all these should eventually be cleaned up. But this is not a stopper for merging!